apps/web/components/ui/UsernameAvailability.tsx

```diff
@@ -0,0 +1,384 @@
+              usernameRef.current.value = currentUsername;
+            }
+          }}>
+          {t("cancel")}
+        </Button>
+      </div>
+    ) : (
+      <></>
+    );
+  };
+
+  const goToBillingPage = async () => {
+    async () => {
+      let url = "";
+      await saveIntentUsername();
+
+      url = "/api/integrations/stripepayment/subscription";
+      const result = await fetch(url, {
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          action: usernameChangeCondition?.toLowerCase(),
+          isPremiumUsername: premiumUsername,
+        }),
+        method: "POST",
+        mode: "cors",
+      });
+      const body = await result.json();
+      window.location.href = body.url;
+    };
+  };
+
+  const saveUsername = () => {
+    if (usernameChangeCondition === UsernameChangeStatusEnum.NORMAL) {
+      updateUsername.mutate({
+        username: inputUsernameValue,
+      });
+    }
+  };
+
+  return (
+    <>
+      <div style={{ display: "flex", justifyItems: "center" }}>
+        <Label htmlFor={"username"}>{t("username")}</Label>
+      </div>
+      <div className="mt-1 flex rounded-md shadow-sm">
+        <span
+          className={classNames(
+            "inline-flex items-center rounded-l-sm border border-gray-300 bg-gray-50 px-3 text-sm text-gray-500"
+          )}>
+          {process.env.NEXT_PUBLIC_WEBSITE_URL}/
+        </span>
+        <div style={{ position: "relative", width: "100%" }}>
+          <Input
+            ref={usernameRef}
+            name={"username"}
+            autoComplete={"none"}
+            autoCapitalize={"none"}
+            autoCorrect={"none"}
+            className={classNames(
+              "mt-0 rounded-l-none",
+              markAsError
+                ? "focus:shadow-0 focus:ring-shadow-0 border-red-500 focus:border-red-500 focus:outline-none focus:ring-0"
+                : ""
+            )}
+            defaultValue={currentUsername}
+            onChange={(event) => setInputUsernameValue(event.target.value)}
+            data-testid="username-input"
+          />
+          {currentUsername !== inputUsernameValue && (
+            <div
+              className="top-0"
+              style={{
+                position: "absolute",
+                right: 2,
+                display: "flex",
+                flexDirection: "row",
+              }}>
+              <span
+                className={classNames(
+                  "mx-2 py-1",
+                  premiumUsername ? "text-orange-500" : "",
+                  usernameIsAvailable ? "" : ""
+                )}>
+                {premiumUsername ? <StarIcon className="mt-[4px] w-6" /> : <></>}
+                {usernameIsAvailable ? <CheckIcon className="mt-[4px] w-6" /> : <></>}
+              </span>
+            </div>
+          )}
+        </div>
+        <div className="xs:hidden">
+          <ActionButtons index="desktop" />
```

Comment:
added index="desktop" so I can search properly while running tests and don't run into dup data-testids

Reply:
This is a particular case where my ActionButtons its being used twice on the view, while one its always hidden according to if the view is desktop or mobile. While doing this I can specifically access the one that runs on the desktop test